src/Net/Http.fs

```diff
@@ -554,20 +554,39 @@ module private HttpHelpers =
         hasContentType.Value
 
     let getResponse (req:HttpWebRequest) silentHttpErrors =
-        if defaultArg silentHttpErrors false then
+
+        let getResponseAsync (req:HttpWebRequest) =
             async {
-                try
+                #if FX_NO_WEBREQUEST_TIMEOUT
                     return! Async.FromBeginEnd(req.BeginGetResponse, req.EndGetResponse)
-                with
-                    | :? WebException as exn -> 
-                        if exn.Response <> null then 
-                           return exn.Response
-                        else 
-                            reraisePreserveStackTrace exn
-                            return Unchecked.defaultof<_>
+                #else
+                    let! child = 
+                        Async.FromBeginEnd(req.BeginGetResponse, req.EndGetResponse)
+                        |> fun x -> Async.StartChild(x, req.Timeout)
```

Comment:
I don't think detecting `req.Timeout` is an option here - it's a has a meaningful default value when not set, so you can't really differentiate based on that. And the thing with "normal synchronous way" here is that it's really the "async way" being run synchronously. This is different to plain old `HttpWebRequest.GetResponse` in that it doesn't take timeout into account, default or not. 

What I can do instead is scrap the idea of using `HttpWebRequest.Timeout` and instead have an optional timeout argument to the `Http` static members at the end of the file. I suppose making the timeout explicit would help more people than it would confuse.